src/server/game/Entities/Player/Player.cpp

```diff
@@ -4237,11 +4239,24 @@ void Player::DeleteFromDB(ObjectGuid::LowType lowGuid, uint32 accountId, bool up
             }
         default:
             LOG_ERROR("entities.player", "Player::DeleteFromDB: Unsupported delete method: {}.", charDelete_method);
+            deleted = false;
             return;
     }
 
+    if (deleted)
+    {
+        const CharacterCacheEntry* cache = sCharacterCache->GetCharacterCacheByGuid(playerGuid);
+        if (cache)
+        {
+            std::string name = cache->Name;
+            sCharacterCache->DeleteCharacterCacheEntry(playerGuid, name);
```

Comment:
```suggestion
            sCharacterCache->DeleteCharacterCacheEntry(playerGuid, cache->Name);
```

Why not this?

Reply:
`DeleteCharacterCacheEntry` takes the string by reference, and erases the cache before using the `name`. So by the point you're using the `name` string it's invalid memory since it's freed just before. By creating a copy of the string before calling `DeleteCharacterCacheEntry` we're safe.

Reply:
Cool, thanks for explaning